apps/web/ee/lib/impersonation/ImpersonationProvider.ts

```diff
@@ -8,56 +8,122 @@ const ImpersonationProvider = CredentialsProvider({
   name: "Impersonation",
   type: "credentials",
   credentials: {
-    username: { label: "Username", type: "text " },
+    username: { type: "text " },
+    teamId: { type: "text" },
   },
   async authorize(creds, req) {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore need to figure out how to correctly type this
     const session = await getSession({ req });
-    if (session?.user.role !== "ADMIN") {
-      throw new Error("You do not have permission to do this.");
-    }
 
     if (session?.user.username === creds?.username) {
       throw new Error("You cannot impersonate yourself.");
     }
 
-    const user = await prisma.user.findUnique({
+    if (!creds?.username) throw new Error("Username must be present");
+
+    // Check session
+    const sessionUserFromDb = await prisma.user.findUnique({
+      where: {
+        id: session?.user.id,
+      },
+      include: {
+        teams: {
+          where: {
+            AND: [
+              {
+                role: {
+                  in: ["ADMIN", "OWNER"],
+                },
+              },
+              {
+                team: {
+                  id: parseInt(creds.teamId),
+                },
+              },
+            ],
+          },
+        },
+      },
+    });
+
+    if (sessionUserFromDb?.role !== "ADMIN" && sessionUserFromDb?.teams.length === 0) {
```

Comment:
What happens when I am an admin but am not an owner of any team? Doesn't this break for non-team impersonation?